daphne/src/vdaf/mod.rs

```diff
@@ -172,17 +173,32 @@ impl VdafConfig {
         version: DapVersion,
     ) -> Result<Report, DapError> {
         let mut rng = thread_rng();
+        let report_extensions = match version {
+            DapVersion::Draft02 => extensions.clone(),
+            _ => vec![],
+        };
         let metadata = ReportMetadata {
             id: ReportId(rng.gen()),
             time,
-            extensions,
+            extensions: report_extensions,
         };
 
         let public_share = Vec::new();
-        let encoded_input_shares = match self {
+        let mut encoded_input_shares = match self {
             Self::Prio3(prio3_config) => prio3_shard(prio3_config, measurement)?,
             Self::Prio2 { dimension } => prio2_shard(*dimension, measurement)?,
         };
+        if version != DapVersion::Draft02 {
+            let mut encoded: Vec<Vec<u8>> = Vec::new();
+            for share in encoded_input_shares.drain(..) {
+                let input_share = PlaintextInputShare {
+                    extensions: extensions.clone(),
```

Comment:
If I'm reading this code correctly, it should be possible to avoid this clone.
```suggestion
                    extensions,
```

Reply:
We're added the passed in extensions into every report, and they are not references in the reports, so we have to clone().  If I take away the clone, it complains that I've moved the value in a previous iteration of the loop.